cleaning up the readme and moving docs into readthedocs (#2291)
redis/redis-py: docs/advanced_features.rst

> Advanced Features

Reviewer: ```suggestion
Advanced features
```

   ...             pipe.watch('OUR-SEQUENCE-KEY')
   ...             # after WATCHing, the pipeline is put into immediate execution
   ...             # mode until we tell it to start buffering commands again.
   ...             # this allows us to get the current value of our sequence
   ...             current_value = pipe.get('OUR-SEQUENCE-KEY')
   ...             next_value = int(current_value) + 1
   ...             # now we can put the pipeline back into buffered mode with MULTI
   ...             pipe.multi()
   ...             pipe.set('OUR-SEQUENCE-KEY', next_value)
   ...             # and finally, execute the pipeline (the set command)
   ...             pipe.execute()
   ...             # if a WatchError wasn't raised during execution, everything
   ...             # we just did happened atomically.
   ...             break
   ...        except WatchError:
   ...             # another client must have changed 'OUR-SEQUENCE-KEY' between
   ...             # the time we started WATCHing it and the pipeline's execution.
   ...             # our best bet is to just retry.
   ...             continue

Note that, because the Pipeline must bind to a single connection for the

> duration of a WATCH, care must be taken to ensure that the connection is

Reviewer: ```suggestion
duration of a WATCH, you need to ensure that the connection is
```

returned to the connection pool by calling the reset() method. If the
Pipeline is used as a context manager (as in the example above) reset()

> will be called automatically. Of course you can do this the manual way

Reviewer: ```suggestion
is called automatically. You can do this manually
```

by explicitly calling reset():

.. code:: pycon

   >>> pipe = r.pipeline()
   >>> while True:
   ...     try:
   ...         pipe.watch('OUR-SEQUENCE-KEY')
   ...         ...
   ...         pipe.execute()
   ...         break
   ...     except WatchError:
   ...         continue
   ...     finally:
   ...         pipe.reset()

> A convenience method named "transaction" exists for handling all the

Reviewer: ```suggestion
A convenience method named "transaction" is used 
```

> boilerplate of handling and retrying watch errors. It takes a callable

Reviewer: ```suggestion
to handle and retry watch errors. It takes a callable
```

that should expect a single parameter, a pipeline object, and any number

> of keys to be WATCHed. Our client-side INCR command above can be written

Reviewer: ```suggestion
of keys to be WATCHed. You can write the client-side INCR command above 
```

like this, which is much easier to read:

.. code:: pycon

   >>> def client_side_incr(pipe):
   ...     current_value = pipe.get('OUR-SEQUENCE-KEY')
   ...     next_value = int(current_value) + 1
   ...     pipe.multi()
   ...     pipe.set('OUR-SEQUENCE-KEY', next_value)
   >>>
   >>> r.transaction(client_side_incr, 'OUR-SEQUENCE-KEY')
   [True]

> Be sure to call pipe.multi() in the callable passed to Redis.transaction

Reviewer: ```suggestion
Prior to any write commands, call pipe.multi() in the callable passed to Redis.transaction.
```

> prior to any write commands.

Reviewer: ```suggestion

```
